pkg/webui/console/containers/top-entities-dashboard-panel/item.js

```diff
@@ -12,18 +12,52 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import React from 'react'
+import React, { useEffect } from 'react'
 import classNames from 'classnames'
+import { useDispatch, useSelector } from 'react-redux'
 
 import { Table } from '@ttn-lw/components/table'
 
 import useBookmark from '@ttn-lw/lib/hooks/use-bookmark'
 import PropTypes from '@ttn-lw/lib/prop-types'
 
+import { startGatewayStatistics, stopGatewayStatistics } from '@console/store/actions/gateways'
+
+import { selectDeviceLastSeen } from '@console/store/selectors/devices'
+import { selectApplicationDerivedLastSeen } from '@console/store/selectors/applications'
+import { selectGatewayStatistics } from '@console/store/selectors/gateways'
+import { selectGatewayLastSeen } from '@console/store/selectors/gateway-status'
+
 import styles from './top-entities-panel.styl'
 
 const EntitiesItem = ({ bookmark, headers, last }) => {
+  const dispatch = useDispatch()
   const { title, ids, path, icon } = useBookmark(bookmark)
+  const entityIds = bookmark.entity_ids
+  const entity = Object.keys(entityIds)[0].replace('_ids', '')
+  const deviceLastSeen = useSelector(state => selectDeviceLastSeen(state, ids.appId, ids.id))
+  const appLastSeen = useSelector(state => selectApplicationDerivedLastSeen(state, ids.id))
+  const statistics = useSelector(selectGatewayStatistics)
+  const gatewayLastSeen = useSelector(selectGatewayLastSeen)
```

Comment:
I don't like that all selectors here are executed regardless of whether they are relevant or not.

```js
let lastSeenSelector
if (entity === 'application') {
  lastSeenSelector = state => selectApplicationDerivedLastSeen(...)
} else if (entity === 'gateway') {
  …
} else if (entity === 'device') {
  …
}

const lastSeen = useSelector(lastSeenSelector)
```